src/v6_indexer/utils.ts

```diff
@@ -480,4 +460,305 @@ export async function getVaultBalances(
+    return 0;
+  }
+}
+
+export async function insertIfNotExistsMarkets(
+  db: DBConnection,
+  proposalAddr: string,
+  daoAddr: string,
+  passBaseMint: string,
+  passQuoteMint: string,
+  failBaseMint: string,
+  failQuoteMint: string
+): Promise<void> {
+  try {
+    // Get DAO to find base and quote mints for spot market
+    const dao = await db.select()
+      .from(schema.v0_6_daos)
+      .where(eq(schema.v0_6_daos.daoAddr, daoAddr))
+      .limit(1);
+    
+    if (dao.length === 0) {
+      logger.warn(`DAO ${daoAddr} not found when initializing markets`);
+      return;
+    }
+
+    const markets = [
+      {
+        daoAddr,
+        proposalAddr: null, // Spot market has no proposal
+        marketType: 'spot',
+        baseMint: dao[0].baseMintAcct,
+        quoteMint: dao[0].quoteMintAcct,
+      },
+      {
+        daoAddr,
+        proposalAddr,
+        marketType: 'pass',
+        baseMint: passBaseMint,
+        quoteMint: passQuoteMint,
+      },
+      {
+        daoAddr,
+        proposalAddr,
+        marketType: 'fail',
+        baseMint: failBaseMint,
+        quoteMint: failQuoteMint,
+      }
+    ];
+
+    await db.insert(schema.futarchy_markets).values(markets).onConflictDoNothing();
+    logger.info(`Initialized markets for proposal ${proposalAddr}`);
+  } catch (error) {
+    logger.error(`Error initializing markets for proposal ${proposalAddr}:`, error);
+  }
+}
+
+/**
+ * Calculate price from pool reserves with decimal adjustment
+ */
+function calculatePriceWithDecimals(
+  quoteReserves: BN,
+  baseReserves: BN,
+  quoteDecimals: number,
+  baseDecimals: number
+): BN {
+  if (baseReserves.isZero() || quoteReserves.isZero()) {
+    return new BN(0);
+  }
+
+  const PRICE_SCALE = new BN(10).pow(new BN(12)); // 1e12
+  const decimalDiff = baseDecimals - quoteDecimals;
+  
+  let adjustedQuoteReserves = quoteReserves;
+  if (decimalDiff !== 0) {
+    const adjustment = new BN(10).pow(new BN(Math.abs(decimalDiff)));
+    if (decimalDiff > 0) {
+      // Base has more decimals, scale quote up
+      adjustedQuoteReserves = quoteReserves.mul(adjustment);
+    } else {
+      // Quote has more decimals, scale quote down
+      adjustedQuoteReserves = quoteReserves.div(adjustment);
+    }
+  }
+  
+  return adjustedQuoteReserves.mul(PRICE_SCALE).div(baseReserves);
+}
+
+/**
+ * Extract and insert prices for specified markets after any event
+ */
+export async function insertIfNotExistsPrices(
+  db: DBConnection,
+  event: LaunchProposalEvent | ConditionalSwapEvent | SpotSwapEvent | ProvideLiquidityEvent | WithdrawLiquidityEvent | FinalizeProposalEvent,
+  proposalAddr: string,
+  slot: BN,
+  marketsToUpdate?: ('spot' | 'pass' | 'fail')[]
+): Promise<void> {
+  try {
+    const futarchyState = event.postAmmState?.state?.futarchy as FutarchyState;
+    if (!futarchyState) {
+      logger.debug('AMM not in futarchy mode, skipping V6 price insertion');
```

Comment:
I understand we're not going to index spot price here yet, just would like t a TODO so we eventually come back to it or we note it in some way.